src/honey_badger/honey_badger.rs

```diff
@@ -126,13 +145,27 @@ where
         self.epoch += 1;
         self.has_input = false;
         let max_epoch = self.epoch + self.max_future_epochs;
-        let mut step = Step::default();
+        let mut msgs = VecDeque::new();
+        // The first message in an epoch announces the epoch transition.
+        msgs.push_back(Target::All.message(MessageContent::EpochStarted.with_epoch(self.epoch)));
+        let mut step = Step::new(Default::default(), Default::default(), msgs);
```

Comment:
This can also just be `Target::All.message(MessageContent::EpochStarted.with_epoch(self.epoch)).into()`.

In general, I'd try to use (and, where necessary, create) as many helper methods for `Step` as possible to avoid using lots of lines in the algorithm code for constructing it.